File: clients/python/src/vibium/check.py
"""Shared result types and semantic Check command."""
from pathlib import Path
from typing import List, Literal, Optional, TypedDict
from .client import BiDiClient
from .model_options import model_params

class CheckEvidence(TypedDict):
    type: Literal["observation"]
    summary: str

class CheckResult(TypedDict):
    status: Literal["passed", "failed", "inconclusive"]
    claim: str
    summary: str
    evidence: List[CheckEvidence]

async def send_check(client: BiDiClient, claim: str, record: Optional[str] = None, context: Optional[str] = None, *, provider: Optional[str] = None, model: Optional[str] = None, base_url: Optional[str] = None, reasoning_effort: Optional[str] = None) -> CheckResult:
    params = {"claim": claim, **model_params(provider, model, base_url, reasoning_effort)}
    if record is not None:
        if not record:
            raise ValueError("record must be a nonempty path")
        params["record"] = str(Path(record).resolve())
    elif context:
        params["context"] = context
    return await client.send("vibium:check.run", params, timeout=210)

File: clients/javascript/src/check.ts
import { ModelOptions, modelParams } from './model-options';
import { resolve } from 'path';
import { BiDiClient } from './bidi';

export interface CheckOptions extends ModelOptions { /** Read-only archive on the runtime host. */ record?: string; }
export interface RecordedCheckOptions extends CheckOptions { record: string; executablePath?: string; }
export interface CheckResult {
  status: 'passed' | 'failed' | 'inconclusive';
  claim: string;
  summary: string;
  evidence: { type: 'observation'; summary: string }[];
}
export const CHECK_TIMEOUT_MS = 210_000;

/** @internal All inference and browser actions run in the existing Go runtime. */
export function sendCheck(client: BiDiClient, claim: string, options: CheckOptions = {}, context?: string): Promise<CheckResult> {
  const params: Record<string, unknown> = { claim, ...modelParams(options) };
  if (options.record !== undefined) {
    if (!options.record) throw new Error('record must be a nonempty path');
    params.record = resolve(options.record);
  } else if (context) params.context = context;
  return client.send<CheckResult>('vibium:check.run', params, CHECK_TIMEOUT_MS);
}
